### DISCUSSION TITLE
.NET MAUI 8.0.7 (Service Release 2)

### DISCUSSION THREAD

**Discussion by davidortinau** [dotnet/maui#20580] [Announcements]
This service release is now available in Visual Studio 17.9.3, and 17.10 Preview 2 on Windows, and the dotnet CLI across all platforms.

The focus of our service releases is resolving high-impact bugs for developers upgrading their applications from Xamarin to .NET 8. 

- [Release notes](https://github.com/dotnet/maui/releases/tag/8.0.7)

**Comment by ziomek64** [comment]
Who cares, images are not even working XD

**Reply by vitorelli** [reply]
Could you please give more details about is not working?

**Reply by ziomek64** [reply]
https://github.com/dotnet/maui/issues/14471

**Reply by ziomek64** [reply]
That issue persists for months since .net 7, was fine on .net6

**Reply by vitorelli** [reply]
Replied on #14471 about your HotReload issue.

**Reply by MAUIoxo** [reply]
Looking at this video isn’t it just a Hot Reload issue? Means when you start it as you did it in the beginning you will have it displayed correctly also when deployed to your real device?

**Reply by ziomek64** [reply]
> Looking at this video isn’t it just a Hot Reload issue? Means when you start it as you did it in the beginning you will have it displayed correctly also when deployed to your real device?

No, it also doesn't work when loading an image in runtime

**Reply by MAUIoxo** [reply]
By „in runtime“ you mean while you are currently debugging your App in Simulator and you change something in a XAML File? This would be a Hot Reload action then. Your App in the Simulator is loaded with you updates on the modified XAML. But this could not be done when your App is deployed and running on a real device. Maybe I did not quite understand your workflow?

**Reply by ziomek64** [reply]
In runtime, changing binded string - source of image in code. I've done that too, with a button or whatever.

**Reply by ziomek64** [reply]
Please check this video @MAUIoxo  - https://github.com/dotnet/maui/issues/14471#issuecomment-1540416516

**Comment by jonmdev** [comment]
I still cannot release anything with Maui due to numerous bugs in basic functions like Border, Label, and Editor:

https://github.com/dotnet/maui/issues/17757 - Editor is completely unusable due to broken scrolling - reported 5 months ago

https://github.com/dotnet/maui/issues/17884 - Android is omitting entire words/letters from Labels - reported 5 months ago

https://github.com/dotnet/maui/issues/18328 - Measurement method for Labels in Android is wrong - likely I suspect reason for truncation above

https://github.com/dotnet/maui/issues/18204 - iOS Borders do not resize correctly - reported 5 months ago

https://github.com/dotnet/maui/issues/18071 - Android does not draw rounded corners on Border correctly - reported 5 months ago

All of these are basic needed functions. How can we make apps without at least working Editor, Label, and Border?

These are not esoteric functions. They will ruin anyone's application. My projects are paralyzed. 5 months later nothing has been fixed. I am just stuck wondering if I will able to release anything ever with Maui after committing all this energy to it. I would appreciate any help or priority in getting these fixed.

**Comment by shadowfoxish** [comment]
Did the release upgrade for my app from 7 to 8, and it went pretty cleanly. There were some changes I had to make that weren't mentioned in the upgrade guide as possible things to look at. Here are my take aways for the upgrade. A few were head-scratchers, but I was able to make it all work with a bit of sleuthing. Hope this helps someone else

- I did appreciate the "compiled binding" warnings showing up, but I was a bit confused that instances of```<FormatString.Spans>``` and some ```<SwipeView><SwipeView.LeftItems/RightItems><SwipeItem binding=... x:DataType=...>``` needed an x:DataType attribute to satisfy the databinding warnings, when the parent controls getting source already had x:DataType specified.
- Visual style change after upgrading; if a ```<Label>``` is nested like so, ```<SwipeView IsEnabled={...}><Grid><Label>``` Maui now appears to apply the ```VisualStateManager.Disabled``` global style for Label from my styles.xaml file. This "cascading" inheritance of the enabled style didn't use to happen. It makes sense, but its different.
- Popups implementing the CommunityToolkit's Popup class stopped rendering content without adding a ```<VerticalStackLayout>```. (Previously, my root most content element for the Popups was ```<Frame>```. After the upgrade, the frame would render, but none of its content would show, until I put a VerticalStackLayout as the first content element of the Popup.
- I had an issue testing on Android device regarding "Fast Deploy", and had to add ```<EmbedAssembliesIntoAPK>``` to the csproj. This change also seems to have the knock-on effect of clearing the app's cache each time, for example, now I have to log in every time I start debugging.
- Had to change a reference from MauiApplication.Current.Services to IPlatformApplication.Current.Services for my IServiceProvider instance.
- Weirdly, ```dotnet clean```, on my build agent no longer works after the upgrade. It complains there's a missing android manifest XML file. I had to disable that build step.
- Another one that got me, and this might just be a .NET 8 thing, but it seems the default .ToString() for decimals has changed. All over my app, I started seeing 3.0000000000000 being printed where previously 3 was shown. If you're using decimals you might need to add StringFormats of 0.# to return to previous functionality if you start seeing a lot of excess decimals printed all over the place.
